src/librustdoc/clean/types.rs

```diff
@@ -740,10 +742,26 @@ impl Attributes {
                             // Extracted #[doc(cfg(...))]
                             match Cfg::parse(cfg_mi) {
                                 Ok(new_cfg) => cfg &= new_cfg,
-                                Err(e) => diagnostic.span_err(e.span, e.msg),
+                                Err(e) => handler.span_err(e.span, e.msg),
                             }
                         } else if let Some((filename, contents)) = Attributes::extract_include(&mi)
                         {
+                            if local {
+                                let mut diag = handler
+                                    .struct_span_warn(attr.span, "`doc(include)` is deprecated");
```

Comment:
Shouldn't this be a lint so that `deny(warnings)` will upgrade it to an error? (cc #80965)

Reply:
I'm not sure how #80965 is related. I'm fine with making this a lint, but I don't know if there's an existing "deprecated feature" lint - I only found deprecations for the standard library.

Reply:
(In general, I don't have much sympathy for `deny(warnings)`, I think it's a bad idea.)

Reply:
> I'm not sure how #80965 is related.

It's related because it contains a deprecation and you originally requested to make it a lint so that `deny(warnings)` would work on it: https://github.com/rust-lang/rust/pull/80965#issuecomment-761839084

> (In general, I don't have much sympathy for `deny(warnings)`, I think it's a bad idea.)

Why is it a bad idea? I find it very useful so that I catch warnings and deprecations in CI. I don't run rustdoc that often locally for my projects, but I find using intra-doc links is a good way of keeping my in-code docs up-to-date since CI will fail when I rename things.

Reply:
> It's related because it contains a deprecation and you originally requested to make it a lint so that deny(warnings) would work on it: #80965 (comment)

Oh well good point :laughing: I'll add a "deprecated nightly feature" lint.

> Why is it a bad idea? I find it very useful so that I catch warnings and deprecations in CI. I don't run rustdoc that often locally for my projects, but I find using intra-doc links is a good way of keeping my in-code docs up-to-date since CI will fail when I rename things.

It's mostly off-topic in this FCP, but - `deny(warnings)` means you're giving up control over whether your code compiles or not. It's better served by `-D warnings`, which allows other people to build your code even if they're using another toolchain.

See also https://github.com/rust-lang/docs.rs/issues/651 - without that change, dozens of crates were not being documented.

Reply:
See https://thephd.github.io/your-c-compiler-and-standard-library-will-not-help-you for why I feel so strongly that new warnings should not be considered a breaking change.

Reply:
> It's mostly off-topic in this FCP, but - `deny(warnings)` means you're giving up control over whether your code compiles or not. It's better served by `-D warnings`, which allows other people to build your code even if they're using another toolchain.

Oh, when I use `deny(warnings)` I mean "deny warnings" in general; i.e., either the attribute or the flag. That explains the confusion :)